src/github.com/getlantern/balancer/balancer.go

```diff
@@ -71,13 +71,19 @@ func (b *Balancer) DialQOS(network, addr string, targetQOS int) (net.Conn, error
 
 	_, port, _ := net.SplitHostPort(addr)
 
-	if port == "" || port == "80" || port == "8080" {
+	// try proxy non-https traffic through trusted servers first
+	if port != "443" {
 		dialers = b.trusted
 	} else {
 		dialers = b.dialers
 	}
 
 	for {
+		// in case dialers == b.trusted but no trusted dialer available,
+		// untrusted dialers are better than no dialer at all.
+		if len(dialers) == 0 {
+			dialers = b.dialers
+		}
```

Comment:
We discussed this a bit here https://github.com/getlantern/lantern/pull/2424#discussion-diff-28866183, probably the best way to fix this is making sure to provide users with (at least one) trusted dialers instead of letting the client pass when no dialers are available.